Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ParticipantsController.cs

```diff
@@ -256,5 +262,46 @@ public ActionResult ResetSubmissions()
 
             return null;
         }
+
+        // TODO: Remove this method after the job is registered
+        public ActionResult DeleteLeftOverFilesInTemp()
+        {
+            var cron = "0 1 * * *";
+            this.backgroundJobs.AddOrUpdateRecurringJob(
+                "DeleteLeftOverFoldersInTemp",
+                () => this.DeleteLeftOverFoldersInTemp(),
+                cron);
+
+            return null;
+        }
+
+        // TODO: Remove this method after the job is registered
+        public void DeleteLeftOverFoldersInTemp()
+        {
+            foreach (var dirPath in Directory.GetDirectories(GlobalConstants.ExecutionStrategyTempPath).ToArray())
+            {
+                var dir = new DirectoryInfo(dirPath);
+                if (dir.Exists && dir.CreationTime < DateTime.Now.AddHours(-1))
+                {
+                    var isDeleted = false;
+                    Stopwatch watch = new Stopwatch();
+                    watch.Start();
+                    while (!isDeleted || watch.Elapsed < TimeSpan.FromSeconds(1))
+                    {
+                        try
+                        {
+                            dir.Delete(true);
```

Comment:
Is this a surefire enough way to delete a directory? What if it contains read-only files? Consider how deletion is done in the DirectoryHelpers class.